### ISSUE TITLE
recent metadata.yaml changes burned my house down

### ISSUE THREAD

**Issue by kwmonroe** [jamesbeedy/layer-puppet-agent#3]
A recent commit brought in `metadata.yaml` with some explicit keys set:

https://github.com/jamesbeedy/layer-puppet-agent/commit/8119439916292a178b5e8de672057102c7d5d681

This hurts because my charm that `includes: ['layer:puppet-agent']` has merged things like `tags`, `requires`, and `series`.  Even worse, this turns my charm that did not include a `subordinate` key into a subordinate charm because puppet-agent's metadata has `subordinate: true`.

Here's what my resultant charm metadata looks like now:

```
$ cat hadoop-namenode/metadata.yaml 
"name": "hadoop-namenode"
"summary": "HDFS master (NameNode) for Apache Bigtop platform"
"maintainer": "Juju Big Data <bigdata@lists.ubuntu.com>"
"description": "Hadoop is a software platform that lets one easily write and run applications\
  \ that process vast amounts of data.\nThis charm manages the HDFS master node (NameNode).\n"
"tags":
- "ops"
- "puppet"
- "cfgmgmt"
- "applications"
- "bigdata"
- "bigtop"
- "hadoop"
- "apache"
"requires":
  "juju-info":
    "interface": "juju-info"
    "scope": "container"
  "datanode":
    "interface": "dfs-slave"
"provides":
  "java":
    "interface": "java"
    "scope": "container"
  "namenode":
    "interface": "dfs"
  "benchmark":
    "interface": "benchmark"
"series":
- "xenial"
- "wily"
- "vivid"
- "trusty"
- "precise"
"subordinate": !!bool "true"
```

The puppet-related tags, juju-info relation, series, and subordinate bits are bad for my charm.  I think the right way forward is to treat puppet-agent as a base layer and drop the metadata.yaml.  We would then create a puppet charm layer that includes puppet-agent and its own metadata.yaml that would be relevant for someone that wants to deploy puppet as a service.

Thoughts?

**Comment by jamesbeedy** [comment]
@kwmonroe Totally, I hadn't realized the metadata got pushed up

**Comment by jamesbeedy** [comment]
fixed in: 0e5482c531b48c9e2218fd707501b6f4255bd64f
